src/TextSegment.cpp

```diff
@@ -14,6 +14,7 @@ TextSegment::TextSegment(const sf::Font& font, std::string text, Markup& setting
   _targetText = text;
   _settings = settings;
   atEnd = false;
+  _needsDraw = true;
```

Comment:
Just use requestDraw?

Reply:
I'm glad you brought that up. :)  Truthfully, I got lazy by the time I got to this class. However, I found it interesting that TextSegment has the same `update` and `draw` functions of the Box and GameSlice classes, but there is no common base class for TextSegment. What do you think about having a base class (perhaps call it `GraphicsBase`?) that has the `draw()`, `needsDraw()`, `redraw()`, and `drawComplete()` functions? Then we can make this and the Box and GameSlice classes all inherit from `GraphicsBase`, and re-use those functions. We might also include the `update()` and `input()` functions in that base class as well.